### ISSUE TITLE
zh-cn: colon in xrefs is dubious

### ISSUE THREAD

**Issue by ghost** [doc-suse-com/l10n#10]
Patrick:

----

Our S. Chinese language reviewer has reported an issue with the syntax of a series of cross-references in the SLED Administration Guides. It is in section 3.1. Here is snippet of the XML, from yast2_gui.xml:

```
  <itemizedlist>
     <listitem>
      <para>
       <xref xrefstyle="select:label quotedtitle" linkend="cha-onlineupdate-you"/>
      </para>
     </listitem>
     [...]
  </itemizedlist>
```
 
The xrefstyle="select:label quotedtitle" attribute is causing a new format, that contains a colon, to be generated. The Chinese reviewer wants us to remove the generated colon.

![image001-1](https://user-images.githubusercontent.com/5476547/130945298-a44ed980-016f-4ee7-8005-e66945014383.png)

I have looked in the common/l10n/zh_cn.xml file, in various levels of fallback, and this format doesn’t seem to exist.

So I have searched elsewhere and found that it may be set by $xref.label-title.separator in the file nwalsh5/1.79.2/common/param.xml, which is used in the file nwalsh5/1.79.2/common/gentext.xsl.

**Comment by ghost** [comment]
I'd suspect this can only be fixed by adding a new gentext key and
modifying gentext.xsl to use that. That would have to be done within
our suse-xsl repo, or alternatively--upstream at the DocBook project.

Given how this is an issue that would presumably affect most people
creating Chinese output with Docbook, pushing it upstream (eventually)
would make most sense, I suspect.

When the translator says "Global issue", how "global" do they
mean?

1. In relation to these specific types of xrefs or do they mean it
   in relation to all our uses of colons in labels and xrefs?
   https://github.com/openSUSE/suse-xsl/blob/main/suse2021-ns/common/l10n/zh_cn.xml#L57,L65

2. If zh-cn is affected, does zh-tw have the same rules?

3. Are ja-jp or ko-kr affected?

**Comment by ptinley-vt** [comment]
1. By "Global issue" here the translator just means that they will not mark every occurrence on the PDF and that the issue noted should be fixed once for all occurrences. Their view would be only what appears in the current PDF. If there were single-byte colons, or colons in general, appearing in other types of xrefs they felt was an incorrect syntax for Chinese, they would mark those, as necessary, too. My feeling is that this is limited to xrefs with attributes similar to 'xrefstyle="select:label quotedtitle"'.

2. Both zh-cn and zh-tw reported this issue.
3. We are checking with Japanese (who didn't report it) and Korean, for which there is no SLED Admin Guide, so no report. I made an KO example using JA text and KO templates. 

I will update here with any feedback we get from JA and KO.

**Comment by ghost** [comment]
Thank you for your efforts in clarifying this!

> If there were single-byte colons, or colons in general, appearing in other types of xrefs they felt was an incorrect syntax for Chinese

I guess that means the spots where we use double-byte colons would still be considered appropriate..? I think that's most everywhere else in the Chinese context.

**Comment by ptinley-vt** [comment]
Yes, double-byte colons are fine. From memory, I don't think there are any other cross-reference types generated in ZH_CN or ZH_TW that contain colons at all.
We see double-byte colons in Figure labels, for example, though I think they are actually PRESENTATION FORM FOR VERTICAL TWO DOT LEADER characters (U+FE30). We've never had a complaint from reviewers about them, so no need to do anything there.

From my experience, for a Chinese reviewer, seeing a single-byte colon in Chinese text is like a red flag. Maybe that is what drew there eye to this issue.

I have queried with our Japanese and Korean linguists and they do not seem to have a problem with the appearance of the single-byte colon in the context of this 'xrefstyle="select:label quotedtitle"' cross-reference, though Korean has asked for an example with better context, which I will follow up with.

**Comment by ptinley-vt** [comment]
I've had further discussion with our Korean language reviewer and they said that, in this context, the colon is **not an issue**.

So, Japanese and Korean can remain using the default behaviour.

**Comment by ghost** [comment]
Thank you! I will try to fix it for both variants of Chinese then!
